### DISCUSSION TITLE
Private key ignored when bootstrapping via SSH

### DISCUSSION THREAD

**Discussion by rombert** [fluxcd/flux2#2205] [Q&A]
Hi,

I am trying to bootstrap fluxcd on a self-hosted private repository. I created a private/public key pair for flux and ensured they have access to the git repo. I am then running something like

```
$ flux bootstrap git --url=ssh://git@my-git-server:2345/org/k8s-flux.git --branch=master --private-key-file=tmp --path=clusters/my-cluster
```

My private key is seemingly ignored and the CLI tries to create a new secret:

```
► cloning branch "master" from Git repository "ssh://git@my-git-server:2345/org/k8s-flux.git"
✔ cloned repository
► generating component manifests
✔ generated component manifests
✔ component manifests are up to date
✔ installed components
✔ reconciled components
► determining if source secret "flux-system/flux-system" exists
► generating source secret
✔ public key: ssh-rsa AAAAB3NzaC1yc2EAAAADAQABAAABgQCpAd8vXDF/YPUlb2/kRKF3bYNVkJ6czefxEI7PCXj6dOqPF6txd2Xkr1VNPsNNtFpDA6CV32MArFYr6EX6NmJnT2mpPywCSIpNvV6HYVGXI6XdigfVImboyYfzQHqYP6HpouP+ZEfoEKphLsCQlcQ3aIC0Reh9DS3ZJ5qm05UwpGEH5ubjGMKciHqlxX1eCSkctXX5tAfDnTMfy5h/Y2j6UC6NaCXWcQ9vD255GYEvyoGX8yqjnowf5qiD2+yrAl1U5B+9O/Ac7DuF5Ac/RcbO5IvFgBMv8miACrpPEWH9Zo2dBKQmP8HMAxJzAEwzNbePhznFBTA+Vs7m2nbZepLgSy30cC9MPzYpIC+022zm6OIq3bVB9Q7WYLnfgRO/pAUUTEKV5FBNM8Trw+9IJ/B6YghKhsx8lKBHbHFqkbGzD5TGnKV/5WZspTPkVVZ865UYO5QxQHDDv40C+pNvaxgbePW3zHukJsrygdamOM1OJjgjTfAIGgnd6YVS2WMcvV0=
✗ Please give the key access to your repository: 
```

I tried placing the pub key next to it ( tmp.pub next to tmp ). I also added it to my SSH agent, and confirmed that it is loaded

```
$ ssh-add -l | grep fluxcd
3072 SHA256:LPblhVUQAhGXaPhrR94IwaGeq+r/c0qqXXvYRIR3nfA fluxcd deploy key (RSA)
```

At this point I am unsure what is wrong. How can I bootstrap fluxcd using a self-hosted git repo and a pre-existing SSH key?

Version/status info:

```
$ flux --version
flux version 0.24.1
$ flux check
► checking prerequisites
✔ Kubernetes 1.21.5 >=1.19.0-0
► checking controllers
✔ helm-controller: deployment ready
► ghcr.io/fluxcd/helm-controller:v0.14.1
✔ kustomize-controller: deployment ready
► ghcr.io/fluxcd/kustomize-controller:v0.18.2
✔ notification-controller: deployment ready
► ghcr.io/fluxcd/notification-controller:v0.19.0
✔ source-controller: deployment ready
► ghcr.io/fluxcd/source-controller:v0.19.2
✔ all checks passed
```

Thanks!

----

Edit: Hm .... so it seems that flux is prompting me to grant access to the public key that was derived from the private key that I supplied. But I instead though that it was asking for a new public key to be granted access.

Maybe the wording can be changed somewhat?

**Comment by kingdonb** [comment]
Thanks for the report! It sounds like the private key was used as you expected, but the output didn't change enough to make it clear that the flag you passed in was really effective.

> `► generating source secret`

Doesn't come out and say "from the private key you provided" but it also doesn't *not say* "with new keys from the random bit bucket"

Do you think it would have helped to add some output here? I hear it's bad practice to present output from user input, but maybe 

> `► generating source secret from key file: 'tmp'`

would have caught your eye and make the next user less likely to spin their wheels wondering if this worked or not?

What change would you suggest now that you're past the issue for a couple of days?

**Reply by rombert** [reply]
Looking back , the prompt is the line that got my attention and the public key listing is what got me confused. I found it confusing because the repository was already cloned with the private key I supplied. (Right?)

Maybe the public key line could be enhanced to look like

> public key (derived from user-supplied private key):  ssh-rsa ...
